tracks/go/exercises/isogram/mentoring.md

```diff
@@ -0,0 +1,76 @@
+# Isogram
+
+This exercise further reinforces concepts introduced in previous exercises, regarding differences between `rune` and `string`.  
+Students are encouraged to find ways to keep track of unique characters in the string.
+
+As always, _Unformatted or unlinted code is a reason not to approve._
+
+### Reasonable solutions
+
+```
+// Package isogram
+package isogram
+
+import "unicode"
+
+// IsIsogram returns true if a word is an isogram
+func IsIsogram(word string) bool {
+	seen := make(map[rune]bool)
+	for _, v := range word {
+		if v == '-' || v == ' ' {
+			continue
+		}
+		v = unicode.ToLower(v)
+		if seen[v] {
+			return false
+		}
+		seen[v] = true
+	}
+	return true
+}
+```
+
+### Common suggestions
+
+**Storage**
+
+- A `map[rune]bool` is a straightforward way to store unique runes.
+  It makes it easier to check for the existence of keys instead of using the longer form:
+
+```
+if _, ok := seen[v]; ok { ... }
+```
+
+- `map[rune]struct{}` uses less memory, but not by a wide margin.
+- `map[rune]int` is an acceptable approach where we return false when the count is more than 1.
+- Using an array for storage is an approach that will considerably improve the performance of the solution:
+
+```
+var alpha [26]bool
+```
+
+However later on in the code, something like the following will need to be adopted:
+
+```
+index := charVal - 'A'
+if alpha[index] {
+	return false
+}
+alpha[index] = true
+```
+
+Working directly with code points this way is both a clever and efficient way to solve the problem, at the expense of readability.  
+If the student goes this route we may suggest the `map[rune]bool` solution, or at the very least to add a comment clarifying their intent.
+
+**Checking characters**
+
+- Since the requirements only mention whitespace and hyphens, checking only for those is acceptable.
+- `unicode.IsLetter()` is otherwise a great alternative that will take care of all invalid/unwanted characters.
+
```

Comment:
It's very common to see code like this:

```go
if unicode.IsLetter(v) {
	... // check map
}
```

This is a nice opportunity to point out the 'early return' pattern:

```go
if !unicode.IsLetter(v) {
	continue
}
... // check map
```

and explain its advantages:

1. The remaining code doesn't need to be indented, improving readability
2. No need for a subsequent `else` clause, because we already returned
3. It's easier to read non-negative `if` conditions, and the action associated with the condition (`continue`) is right next to the condition itself in the source code, instead of many lines later.

Reply:
Thanks @bitfield , I addressed all your points.